readthedocs/telemetry/collectors.py

```diff
@@ -89,8 +90,56 @@ def collect(self):
                 "all": all_apt_packages,
             },
         }
+        data["doctool"] = {
+            "name": self._get_doctool_name(),
+            "extensions": self._get_doctool_extensions(),
+            "theme": self._get_doctool_theme(),
+        }
         return data
 
+    def _get_doctool_name(self):
+        if self.config.sphinx:
+            return "sphinx"
+
+        if self.config.mkdocs:
+            return "mkdocs"
+
+        return "generic"
+
+    def _get_doctool_extensions(self):
+        if self._get_doctool_name() != "sphinx":
+            return []
+
+        code, stdout, _ = self.run(
+            "python",
+            "-c",
+            "import conf; import json; print(json.dumps(conf.extensions))",
```

Comment:
this will execute everything from that file again, producing undesired effects on some projects. We should probably find a way to get this information from our extension.

Reply:
This feels like something that could break randomly and in confusing ways. I don't think we should be running users code like this outside of the normal build process. I agree with a lightweight approach, but this feels like something that will likely have side effects and we should be more conservative here.

Reply:
> I don't think it will produce undesired effects on projects since they are already built and the HTML are already prepared to be uploaded.

I guess it depends on what users are doing, they could be calling an external process, delete things to have a "clean" directory before building, hitting external APIs.

>  How do you think we can use our extension to get this data? I mean, it's available during build time and we have control there, but we need to output it somehow so we can read it while collecting BuildData.

The simple approach would be to output the relevant information into a file (a json file maybe?), and then read that file in the build telemetry step.

Reply:
I opened this PR that does the dump of the wanted settings from the `readthedocs-sphinx-ext`: https://github.com/readthedocs/readthedocs-sphinx-ext/pull/117